### PR TITLE
feat: openBox now works with IdentityWallet instance

### PR DESCRIPTION
**Pull request by oed**
This PR adds support for passing an IdentityWallet instance to `openBox`. 

This functionality still needs some integration tests to make sure that all IdentityWallet functionality works as intended.

### CODE REVIEW

File: src/3box.js [hunk 1]
```diff
@@ -96,12 +96,16 @@ class Box {
...
+    }
+
     this.public = new PublicStore(this._3id.muportFingerprint + '.public', this._linkProfile.bind(this), this.replicator, this._3id)
     this.private = new PrivateStore(this._3id.muportFingerprint + '.private', this.replicator, this._3id)
-    await Promise.all([
-      this.public._load(),
-      this.private._load()
-    ])
+    await this.public._load()
+    await this.private._load()
```
Comment (zachferland) [inline src/3box.js]: prior implementation would have been faster in most env, unless you need public load to resolve before private load

File: src/__tests__/idWallet.integration.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  it('should get same state on openBox with IdentityWallet opened using second authSecret', async () => {
+    publishHasEntries()
+    idWallet = new IdentityWallet({ authSecret: AUTH_2, ethereumAddress: ADDRESS })
+    const box = await Box.openBox(idWallet, mockEthProvider, opts)
+    await box.syncDone
+    expect(await box.public.all()).toEqual(pubState)
+    await box.close()
+  })
+})
```
Comment (zachferland) [inline src/__tests__/idWallet.integration.js]: 👍

File: src/__tests__/testUtils.js [hunk 3]
```diff
@@ -1,178 +1,37 @@
...
+      Addresses: {
+        Swarm: [
+          `/ip4/127.0.0.1/tcp/${4004 + id * 2}`,
+          `/ip4/127.0.0.1/tcp/${4005 + id * 2}/ws`
+        ],
+        API: `/ip4/127.0.0.1/tcp/${5003 + id}`,
+        Gateway: `/ip4/127.0.0.1/tcp/${9091 + id}`
+      },
+      Bootstrap: []
+    }
```
Comment (zachferland) [inline src/__tests__/testUtils.js]: yes! thanks

File: src/3box.js (hunk 1 above)
Comment (oed) [reply src/3box.js]: I was seeing some weird concurrency issue here when running the integration test. I don't think this would be much slower at all tbh!

**Comment by oed** [comment]
Needs a rebase, then should be ready for merge.

**Pull request merged**